redis/src/bloom.rs

```diff
@@ -0,0 +1,310 @@
+    pub number_of_items_inserted: i64,
+    /// The expansion rate of the filter (if filter is scaling)
+    pub expansion_rate: Option<i64>,
+}
+
+impl FromRedisValue for AllInfoResponse {
+    fn from_redis_value(v: &Value) -> RedisResult<Self> {
+        let hm: RedisResult<HashMap<String, Option<_>>> = from_redis_value(v);
+        hm.map(|hm| {
+            let resp = &mut AllInfoResponse {
+                capacity: 0,
+                memory_size: 0,
+                number_of_filters: 0,
+                number_of_items_inserted: 0,
+                expansion_rate: None,
+            };
+
+            hm.iter().for_each(|(k, v)| match k.as_str() {
+                "Capacity" => resp.capacity = v.map_or(0, |x| x),
+                "Size" => resp.memory_size = v.map_or(0, |x| x),
+                "Number of filters" => resp.number_of_filters = v.map_or(0, |x| x),
+                "Number of items inserted" => resp.number_of_items_inserted = v.map_or(0, |x| x),
+                "Expansion rate" => resp.expansion_rate = *v,
+                _ => { /*quietly ignore unknown keys*/ }
+            });
+            *resp
+        })
+    }
+}
+/// Response returned by the bf_info command (used to return a specific field).
+/// See the [Redis documentation](https://redis.io/docs/latest/commands/bf.info/) for more information.
+///
+#[derive(Debug)]
+pub struct SingleInfoResponse {
+    /// The value of the requested field if exists
+    pub value: Option<i64>,
+}
+
+macro_rules! not_convertible_error {
+    ($v:expr, $det:expr) => {
+        RedisError::from((
+            ErrorKind::TypeError,
+            "Response type not convertible",
+            format!("{:?} (response was {:?})", $det, $v),
+        ))
+    };
+}
+
+impl FromRedisValue for SingleInfoResponse {
+    fn from_redis_value(v: &Value) -> RedisResult<Self> {
+        match v {
+            Value::Array(vec) => {
+                let resp = SingleInfoResponse {
+                    value: vec.first().and_then(|v| match v {
+                        Value::Int(i) => Some(*i),
+                        _ => None,
+                    }),
+                };
+                Ok(resp)
+            }
+            Value::Map(map) => {
+                let resp = SingleInfoResponse {
+                    value: map.first().and_then(|(_, v)| match v {
+                        Value::Int(i) => Some(*i),
+                        _ => None,
+                    }),
+                };
+                Ok(resp)
+            }
+
+            _ => Err(not_convertible_error!(
+                v,
+                "expected vector (resp2) or map (resp3)"
+            )),
+        }
+    }
+}
+
+/// The result of bf_scan_dump
+/// See the [Redis documentation](https://redis.io/docs/latest/commands/bf.scandump/) for more information.
+#[derive(Debug, Clone)]
+pub struct DumpResult {
+    /// The iterator id to be used in the next call to `bf_scan_dump`
+    pub iterator: i64,
```

Comment:
does this need to be public?

Reply:
it does, since the corresponding BF.LOADCHUNK command, uses these id's when restoring the bloom filter.